src/runtime/virtcontainers/qemu_arch_base.go

```diff
@@ -28,6 +28,12 @@ type qemuArch interface {
 	// disableNestingChecks nesting checks will be ignored
 	disableNestingChecks()
 
+	 // Enables memory encryption for this architecture
+	enableMemEncrypt()
+
+	// Disable memory encryption for this architecture
+	disableMemEncrypt()
```

Comment:
Do we really need a disabling switch? Can we assume it's disabled by default? I think the code here is a NOP when `memEncrypt` is false? And it'd be the same for both TDX and SE afaict.

Reply:
> I believe this is definitely doable on top of #1590. Maybe it would make sense to rebase on top of that PR so that:
> 
>     1. We can practically verify that it would work for SEV as well
> 
>     2. It would be easy to either merge both at the same time or merge SEV (and SE) support as follow up PRs from #1590
> 
> 
> WDYT?

Thanks, @sameo. I Totally agree. #1590 has done a lot more as far as detecting and isolating the architecture-specific options (opposed to the global settings of this PR). It makes sense to rebase this PR onto the structure introduced in #1590. 

In today's meeting, I suggested splitting #1590 into two PRs, one with the confidential-guest framework and the other with the TDX-specific additions. Have you any further thoughts on this?

Reply:
@devimc is going to split his top patch into 2: a framework and a TDX specific one.

Reply:
Great. I'm currently working to rebase this with @devimc patch